agr_literature_service/lit_processing/oneoff_scripts/cleanup_resource_data.py

```diff
@@ -0,0 +1,157 @@
+import logging
+
+from agr_literature_service.lit_processing.utils.sqlalchemy_utils import create_postgres_session
+from agr_literature_service.api.models import ReferenceModel, EditorModel, ResourceModel
+
+logging.basicConfig(format='%(message)s')
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def fix_resource_data():
+
+    db_session = create_postgres_session(False)
+
+    logger.info("Getting data from the database...")
+
+    resource_id_to_nlm = {}
+    rows = db_session.execute("SELECT resource_id, curie FROM cross_reference "
+                              "WHERE curie_prefix = 'NLM' and is_obsolete = False").fetchall()
+    for x in rows:
+        resource_id_to_nlm[x[0]] = x[1]
+
+    duplicate_resource_id_to_correct_resource_id = get_duplicate_resources(db_session,
+                                                                           resource_id_to_nlm)
+
+    logger.info("Updating resource_ids in EDITOR table...")
+
+    # update_editor_table(db_session, duplicate_resource_id_to_correct_resource_id)
```

Comment:
I see the code exists for update_editor_table and update_reference_table, I'm not sure if it's commented out because it shouldn't be run, or to treat the review as if it's going to be run but it was commented out in testing.
I would think you want to delete all the editors that exist in duplicate resources, but that you'd want to just remove the reference-resource connection for duplicate resources.
I think right now it's doing the same thing for both, mapping editors and references to the new resource, but I don't think we need to.  An editor should only exist for a given resource, not multiple resources (as opposed to a person that could be connected to both), so I think it can be safely deleted.  For reference-resource connections, I would think that the reference is already connected to the correct resource and doesn't need to be reconnected, but I haven't looked at the data, but I guess it'd be good to check whether it's making a duplicate reference-resource connection (I don't even know if that's possible, it might not be, in which case it's probably fine).

Reply:
@azurebrd oops good catch!! lol 

Just commented out the two update statements.

Reply:
This is solved